shell/platform/darwin/ios/framework/Source/FlutterAppDelegate.mm

```diff
@@ -92,10 +92,13 @@ - (void)application:(UIApplication*)application
       didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
 }
 
+#pragma GCC diagnostic push
```

Comment:
This overall looks good, are these the best we can do though?  Would it make any sense to just conditionally compile them instead based on the SDK level?

Reply:
I don't want the symbol to disappear if they change the deployment target.  As of March, all apps submitted to the App Store must use iOS SDK 12, and the flutter create template defaults to iOS 8.  Since the replacement APIs aren't drag-and-drop, it will probably take some thought for developers to move off of them, and I don't want them to rush the process because they need to make an App Store submission to, say, fix a critical bug, but they can't compile because this symbol is gone.  I'd rather pass the buck of behavior changes off to UIKit and give developers a warning that they need to swap it out.

Deprecating in the header and suppressing the deprecation warning in the implementation is the strange but standard pattern.  :-(